client/my-sites/media-library/media-file.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getUrlParts } from 'lib/url/url-parts';
+import isPrivateSite from 'state/selectors/is-private-site';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import getSelectedSiteId from 'state/ui/selectors/get-selected-site-id';
+import getSelectedSiteSlug from 'state/ui/selectors/get-selected-site-slug';
+import ProxiedImage, { RenderedComponent } from './proxied-image';
+
+const parseMediaURL = ( url: string, siteSlug: string ) => {
```

Comment:
Should this get pulled out into a media / utils file or something? 

https://github.com/Automattic/wp-calypso/blob/1c7d5320a2ce91a316d2ed0d8e4f97bf06d79e8d/client/lib/media/utils.js

Also, while I'm thinking about it, this can be memoized, right?

Reply:
Yes and yes :-)